src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -78,6 +78,11 @@ public function __construct(array $normalizers = array(), array $encoders = arra
             if ($normalizer instanceof NormalizerAwareInterface) {
                 $normalizer->setNormalizer($this);
             }
+
+            if (!($normalizer instanceof NormalizerInterface || $normalizer instanceof DenormalizerInterface)) {
+                @trigger_error(\sprintf('Passing normalizers (%s) which do not implement either %s or %s has been deprecated since 4.2 and will throw an exception in 5.0', get_class($normalizer), NormalizerInterface::class, DenormalizerInterface::class), E_USER_DEPRECATED);
+                // throw new \InvalidArgumentException(\sprintf('The class %s does not implement %s or %s', get_class($normalizer), NormalizerInterface::class, DenormalizerInterface::class));
```

Comment:
please add double quotes around all the "%s"
the " and will throw an exception in 5." part should be removed
and a final dot added

Reply:
@nicolas-grekas removing the 5.0 part is not done in other places, btw? https://github.com/symfony/symfony/pull/26564/files#diff-9d63a61ac1b3720a090df6b1015822f2R729

Reply:
it should have been :)